torch/distributed/_fsdp/flatten_params_wrapper.py

```diff
@@ -127,10 +113,20 @@ def _flatten_params(self) -> None:
         attributes with views to the flat param.
         """
         # register the flatten one
-        assert self.flat_param is not None, (
-            "Can not flatten params when flat_param is None"
+        params = self._init_flatten_params()
+        flat_param = nn.Parameter(
+            torch.cat(
+                [
+                    p.detach().reshape(-1)
+                    if isinstance(p, nn.Parameter)
+                    else p.reshape(-1)
+                    for p in params
+                ],
+                0,
+            ),
+            requires_grad=params[0].requires_grad,
         )
-        self.register_parameter("flat_param", self.flat_param)
+        self.register_parameter("flat_param", flat_param)
```

Comment:
curious, any reason for not directly using `self.flat_param = nn.Parameter(..)` and remove this line? Looks like it will do the same?

https://github.com/pytorch/pytorch/blob/191b48b12f33e1e9525882da0c62b68686d69e42/torch/nn/modules/module.py#L1198-L1203